### PR TITLE
owner(ticdc): decouple feed state manager

### LINKED ISSUE #10275: owner(ticdc): decouple feed state manager
decouple feed state manager

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve?

Issue Number: close #10275

### What is changed and how it works?
 add an interface to decouple feed manager 

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: pkg/orchestrator/reactor_state.go [hunk 1]
```diff
@@ -203,6 +203,223 @@ func NewChangefeedReactorState(clusterID string,
...
+	return s.Info
+}
+
+// GetChangefeedStatus returns the changefeed status.
+func (s *ChangefeedReactorState) GetChangefeedStatus() *model.ChangeFeedStatus {
+	return s.Status
+}
+
+// SetWarning sets the warning to changefeed
+func (s *ChangefeedReactorState) SetWarning(lastError *model.RunningError) error {
```
Comment (3AceShowHand) [inline pkg/orchestrator/reactor_state.go:222]: always return nil, no returned value

File: pkg/orchestrator/reactor_state.go [hunk 2]
```diff
@@ -203,6 +203,223 @@ func NewChangefeedReactorState(clusterID string,
...
+			return nil, false, nil
+		}
+		info.Warning = lastError
+		return info, true, nil
+	})
+	return nil
+}
+
+// SetError sets the error to changefeed
+func (s *ChangefeedReactorState) SetError(lastError *model.RunningError) error {
```
Comment (3AceShowHand) [inline pkg/orchestrator/reactor_state.go:234]: ditto, no need to return any value.

File: pkg/orchestrator/reactor_state.go [hunk 3]
```diff
@@ -203,6 +203,223 @@ func NewChangefeedReactorState(clusterID string,
...
+		}
+		info.Error = lastError
+		return info, true, nil
+	})
+	return nil
+}
+
+// RemoveChangefeed removes the changefeed and clean the information and status.
+func (s *ChangefeedReactorState) RemoveChangefeed() error {
+	// remove info
```
Comment (3AceShowHand) [inline pkg/orchestrator/reactor_state.go:247]: ditto

File: pkg/orchestrator/reactor_state.go [hunk 4]
```diff
@@ -203,6 +203,223 @@ func NewChangefeedReactorState(clusterID string,
...
+	result := make([]*model.RunningError, 0, len(runningErrors))
+	for _, err := range runningErrors {
+		result = append(result, err)
+	}
+	return result
+}
+
+// TakeProcessorWarnings reuturns the warning of the changefeed and clean the warning.
+func (s *ChangefeedReactorState) TakeProcessorWarnings() []*model.RunningError {
+	var runningWarnings map[string]*model.RunningError
```
Comment (3AceShowHand) [inline pkg/orchestrator/reactor_state.go:339]: This method looks can be further simplified

**Review by 3AceShowHand — APPROVED** [review]

File: pkg/orchestrator/reactor_state.go [hunk 5]
```diff
@@ -203,6 +203,223 @@ func NewChangefeedReactorState(clusterID string,
...
+				zap.String("namespace", s.ID.Namespace),
+				zap.String("changefeed", s.ID.ID),
+				zap.Any("oldCheckpointTs", oldCheckpointTs),
+				zap.Any("newCheckpointTs", status.CheckpointTs),
+			)
+			return status, true, nil
+		}
+		return status, false, nil
+	})
+	return nil
```
Comment (asddongmen) [inline pkg/orchestrator/reactor_state.go:301]: ditto

File: pkg/orchestrator/reactor_state.go [hunk 6]
```diff
@@ -203,6 +203,223 @@ func NewChangefeedReactorState(clusterID string,
...
+				log.Info("update changefeed epoch",
+					zap.String("namespace", s.ID.Namespace),
+					zap.String("changefeed", s.ID.ID),
+					zap.Uint64("perviousEpoch", previous),
+					zap.Uint64("currentEpoch", info.Epoch))
+			}
+		}
+		return info, changed, nil
+	})
+	return nil
```
Comment (asddongmen) [inline pkg/orchestrator/reactor_state.go:420]: ditto

**Review by asddongmen — APPROVED** [review]

**Pull request merged**